lib/posix/semaphore.c

```diff
@@ -15,7 +15,7 @@
 struct nsem_obj {
 	sys_snode_t snode;
 	sem_t sem;
-	unsigned int ref_count;
+	int ref_count;
```

Comment:
@npitre I've modified other part that uses `ref_count` to `int` as well, and changed the overflow test from `__ASSERT_NO_MSG(nsem->ref_count != UINT_MAX)` to `__ASSERT_NO_MSG(nsem->ref_count != INT_MAX)`